insights/parsers/krb5.py

```diff
@@ -0,0 +1,132 @@
+"""
+krb5 Configuration file - File
+==============================
+krb5 Configuration are ``/etc/krb5.conf`` and ``/etc/krb5.conf.d/*``,
+and the content format is similar to ``INI config``, but they include
+values that span multiple lines. Multi-line values start with a '{'
+and end with a '}', and we join them together by setting the `is_squ`
+variable to True while in a multi-line value.
+
+The content of file ``/etc/krb5.conf`` and ``/etc/krb5.conf.d/*`` is
+recorded via INI format.
+
+Typical output of the ``/etc/krb5.conf`` is::
+
+Example:
+    >>> krb5_content = '''
+    [realms]
+     dns_lookup_realm = false
+     ticket_lifetime = 24h
+     default_ccache_name = KEYRING:persistent:%{uid}
+     EXAMPLE.COM = {
+      kdc = kerberos.example.com
+      admin_server = kerberos.example.com
+     }
+     89EXAsdcsdMPLE.COM = {
+      kdc = ks2321serberos.example.com
+      admin_server = kc5433scderberos.example.com
+     }
+     [libdefaults]
+      dnsdsd = false
+      tilnvs = 24h
+      89EXAsd = {
+       kdc = ks2321serberos.exampsdcsm
+       admin_server = kc5433scderberos.exasdscdm
+     }
+    # renew_lifetime = 7d
+    # forwardable = true
+    # rdns = false
+    '''.strip()
+
+    >>> from insights.tests import context_wrap
+    >>> shared = {Krb5Configuration: Krb5Configuration(context_wrap(krb5_content))}
+    >>> krb5_info = shared[Krb5Configuration]
+    >>> krb5_info.data["libdefaults"]["dnsdsd"]
+    "false"
+    >>> krb5_info.data["realms"]["EXAMPLE.COM"]["kdc"]
+    "kerberos.example.com"
+"""
+
+
+from insights.core import Parser, LegacyItemAccess
+from insights.core.plugins import parser
+from .. import get_active_lines
+
+
+class Krb5ConfigParser(Parser, LegacyItemAccess):
+    def parse_content(self, content):
+        dict_all = {}
+        is_squ = False
+        section_name = ""
+        squ_value = {}
+        squ_section_name = ""
+        section_value = {}
+        for line in get_active_lines(content):
+            line = line.strip()
+            if is_squ:
+                # If in {} sub_section, get the key_value pair
+                if "=" in line:
+                    key, value = [i.strip() for i in line.strip().split('=', 1)]
```

Comment:
This and the following `line.strip()` can be replaced with `line` because of line-65 did it.